### DISCUSSION TITLE
How to use protobuf encoder/decoder

### DISCUSSION THREAD

**Discussion by JonasHiltl** [nats-io/nats.go#919] [Q&A] [answered]
I have a simple `Connect` function which should connect to a Nats cluster and return an `EncodedConn`. 
It is straight forward to specify json encoding by just passing the `nats.JSON_ENCODER` to the `NewEncodedConn` function. But I want to use protobuf encoding/decoding and it looks like the [Encode/Decode](https://github.com/nats-io/nats.go/blob/main/encoders/protobuf/protobuf_enc.go) function is already implement. I just couldn't find a way to specify protobuf encoding in for the `NewEncodedConn` function. 
How can I specify the Protobuf encoder in the  `NewEncodedConn` function?

Just for reference, this is my basic function to connect to Nats:
```go
func Connect() (*nats.EncodedConn, error) {
	cluster, exists := os.LookupEnv("NATS_CLUSTER")
	if !exists {
		return nil, errors.New("nats connection url not defined")
	}

	nc, err := nats.Connect(cluster)
	if err != nil {
		return nil, err
	}
	c, err := nats.NewEncodedConn(nc, "protobuf")
	if err != nil {
                // no encoder registered for 'protobuf'
		return nil, err
	}
	return c, nil
}
```

**Comment by derekcollison** [comment, marked as answer]
https://github.com/nats-io/nats.go/blob/main/test/protobuf_test.go
